### DISCUSSION TITLE
Phase field crack

### DISCUSSION THREAD

**Discussion by caoductoan379** [idaholab/moose#16282] [Q&A General] [answered]
my input file is  for phase field crack. I got an error below. Could you please let me know what is problem by using  "MeshModifiers" ? 

The error i got as follow: 

*** ERROR ***
Cyclic_2D_S1.i:8: section 'MeshModifiers' does not have an associated "Action".
Did you misspell it?
Cyclic_2D_S1.i:9: section 'MeshModifiers/noncrackBottom2' does not have an associated "Action".
Did you misspell it?
Cyclic_2D_S1.i:15: section 'MeshModifiers/noncrackBottom1' does not have an associated "Action".
Did you misspell it?

[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
0x00007fbd1d1c0457 in __GI___waitpid (pid=4544, stat_loc=stat_loc@entry=0x7ffc63617da8, options=options@entry=0) at ../sysdeps/unix/sysv/linux/waitpid.c:30
#0  0x00007fbd1d1c0457 in __GI___waitpid (pid=4544, stat_loc=stat_loc@entry=0x7ffc63617da8, options=options@entry=0) at ../sysdeps/unix/sysv/linux/waitpid.c:30
#1  0x00007fbd1d12b177 in do_system (line=<optimized out>) at ../sysdeps/posix/system.c:149
#2  0x00007fbd1de9b6ac in libMesh::print_trace(std::ostream&) () from /home/user/miniconda3/envs/moose/libmesh/lib/libmesh_opt.so.0
#3  0x00007fbd20290833 in moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) () from /projects/moose/framework/libmoose-opt.so.0
#4  0x00007fbd1fe33050 in void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) () from /projects/moose/framework/libmoose-opt.so.0
#5  0x00007fbd1ff29baf in Parser::parse(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /projects/moose/framework/libmoose-opt.so.0
#6  0x00007fbd202c2139 in MooseApp::setupOptions() () from /projects/moose/framework/libmoose-opt.so.0
#7  0x00007fbd2029e539 in MooseApp::run() () from /projects/moose/framework/libmoose-opt.so.0
#8  0x00005631cff7b1ec in main ()
[Inferior 1 (process 4543) detached]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

If you can help me for that, I really appreciate your help .

Kind regards,
Toan D Cao
University of Alabama
Department of Mechanical Engineering

**Comment by GiudGiud** [comment, marked as answer]
Hi

MeshModifiers are now deprecated. Please use MeshGenerators instead. 

Best,
Guillaume

**Reply by caoductoan379** [reply]
Hi Guillaume
Thanks for your help. if i have this code, what i can change in this code ?

[Mesh]
  type = EBSDMesh
  filename = Syn2_2D.txt
  displacements = 'disp_x disp_y'
[]

[MeshModifiers]

  [./noncrackBottom2]
    type = BoundingBoxNodeSet
    new_boundary = noncrackBottom2
    bottom_left = '0 0 0'
    top_right = '1 0 0'
  [../]
  [./noncrackBottom1]
    type = BoundingBoxNodeSet
    new_boundary = noncrackBottom1
    bottom_left = '9.5 0 0'
    top_right = '10 0 0'
  [../]

[]

**Reply by aeslaughter** [reply]
MeshModifiers have been removed, so this will no longer run with current MOOSE.

**Reply by caoductoan379** [reply]
Hi 

Thanks for your feedback.

**Comment by caoductoan379** [comment]
Thanks, I solved this problem by using MeshGenerators.
